### PR TITLE
Rebase PR #678

### PR DESCRIPTION
**Pull request by yeganer**
This is a rebase onto the current master of #678.

### CODE REVIEW

File: tardis/montecarlo/montecarlo.pyx [hunk 1]
```diff
@@ -11,12 +11,14 @@ from numpy cimport PyArray_DATA
 from astropy import constants
 from astropy import units
 from libc.stdlib cimport free
+from tardis.util import intensity_black_body
```
Comment (unoebauer) [inline tardis/montecarlo/montecarlo.pyx]: Is this needed?

**Comment by yeganer** [comment]
You found the one line that I failed to clean up, good job!

Ulrich Noebauer <notifications@github.com> schrieb am Mi., 17. Mai 2017 um
14:40 Uhr:

File: tardis/simulation/base.py [hunk 2]
```diff
@@ -410,3 +414,161 @@ def from_config(cls, config, **kwargs):
                    convergence_strategy=config.montecarlo.convergence_strategy,
                    nthreads=config.montecarlo.nthreads)
 
+    def make_source_function(self):
+        """
+        Calculates the source function using the line absorption rate estimator `Edotlu_estimator`
+
+        Formally it calculates the expresion ( 1 - exp(-tau_ul) ) S_ul but this product is what we need later,
```
Comment (talytha) [inline tardis/simulation/base.py]: "expresion" -> "expression"

File: tardis/simulation/base.py [hunk 3]
```diff
@@ -410,3 +414,161 @@ def from_config(cls, config, **kwargs):
...
+        att_S_ul, Edot_u = self.make_source_function()
+        J_blues    = plasma.j_blues
+        J_rlues    = plasma.j_blues * np.exp( -plasma.tau_sobolevs.values) + att_S_ul
+
+        r_shells = np.zeros((num_shell+1,1))
+        # Note the reorder from outer to inner
+        r_shells[1:,0],r_shells[0,0] = self.runner.r_inner_cgs[::-1] / R_max, 1.0
+        z_crossings = np.sqrt(r_shells**2 - ps**2)
+
+        z_ct = z_crossings/ct
```
Comment (wkerzendorf) [inline tardis/simulation/base.py]: pep8

Comment (unoebauer) [inline tardis/simulation/base.py]: Really ;)?

Comment (yeganer) [inline tardis/simulation/base.py]: I can fix all PEP8 violations introduced by this PR or I could work on the proper version which removes 80% of the code introduced by this PR anyway.

I'm perfectly fine fixing PEP8 but I don't think we gain much for the time this takes.

**Comment by wkerzendorf** [comment]
@yeganer sounds good. merging

**Pull request merged**